arb_os/codeSegment.mini

```diff
@@ -340,7 +340,7 @@ impure func generateCodeForEvmSegment(
                             }
                         } else {
                             if (opcode == 0x45) { // GASLIMIT
-                                code = pushValue(getGasLimitConstant(), code);
+                                code = pushValue(getGasLimitConstant(), code);  // return the chain's per-tx gas limit
```

Comment:
I think this should use the dynamic per block gas limit instead of the transaction gas limit